packages/calling-component-bindings/src/providers/CallProvider.tsx

```diff
@@ -57,5 +60,26 @@ export const CallProvider = (props: CallProviderProps): JSX.Element => <CallProv
  * @public
  */
 export const useCall = (): Call | undefined => {
-  return useContext(CallContext)?.call;
+  const call = useContext(CallContext)?.call;
+  if (call && !isACSCall(call)) {
+    throw new Error('TeamsCall object was provided, try useTeamsCall() instead');
+  }
+  return call;
+};
+
+/* @conditional-compile-remove(teams-identity-support) */
+/**
+ * Hook to obtain {@link @azure/communication-calling#TeamsCall} from the provider.
+ *
+ * Useful when implementing a custom component that utilizes the providers
+ * exported from this library.
+ *
```

Comment:
Let's add to this documentation that you must have previously used the `CallProvider` with a `TeamsCall` to use this hook.

Same for the comment for `useCall`.